Trying to build this with debian tools some compilation warnings.

I don't expect too much and I'll see if I can fix these myself.

Here are the relevant parts of  `labwc-tweaks-gtk_0.1.0-1_amd64.build`

<details>
<summary>build log</summary>

```
dpkg-buildpackage -us -uc -ui
dpkg-buildpackage: info: source package labwc-tweaks-gtk
dpkg-buildpackage: info: source version 0.1.0-1
dpkg-buildpackage: info: source distribution UNRELEASED
dpkg-buildpackage: info: source changed by Mick Amadio <01micko@gmail.com>
 dpkg-source --before-build .
dpkg-buildpackage: info: host architecture amd64
dpkg-source: info: using options from labwc-tweaks-gtk-0.1.0/debian/source/options: --tar-ignore=*.deb --tar-ignore=.git* --tar-ignore=install --extend-diff-ignore=.git*
 debian/rules clean
dh_auto_configure -Smeson+ninja
	cd obj-x86_64-linux-gnu && DEB_PYTHON_INSTALL_LAYOUT=deb LC_ALL=C.UTF-8 meson setup .. --wrap-mode=nodownload --buildtype=plain --prefix=/usr --sysconfdir=/etc --localstatedir=/var --libdir=lib/x86_64-linux-gnu -Dpython.bytecompile=-1
The Meson build system
Version: 1.4.0
Source dir: /home/mick/packages/deb-create_tweaks_gtk/work_mkdeb/labwc-tweaks-gtk-0.1.0
Build dir: /home/mick/packages/deb-create_tweaks_gtk/work_mkdeb/labwc-tweaks-gtk-0.1.0/obj-x86_64-linux-gnu
Build type: native build
Project name: labwc-tweaks
Project version: 0.1.0
C compiler for the host machine: cc (gcc 13.2.0 "cc (Debian 13.2.0-24) 13.2.0")
C linker for the host machine: cc ld.bfd 2.42
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: YES (/usr/bin/pkg-config) 1.8.1
Run-time dependency gtk+-3.0 found: YES 3.24.41
Run-time dependency libxml-2.0 found: YES 2.9.14
Run-time dependency gio-2.0 found: YES 2.80.2
Build targets in project: 1

labwc-tweaks 0.1.0

  User defined options
    buildtype         : plain
    libdir            : lib/x86_64-linux-gnu
    localstatedir     : /var
    prefix            : /usr
    sysconfdir        : /etc
    wrap_mode         : nodownload
    python.bytecompile: -1

Found ninja-1.11.1 at /usr/bin/ninja
dh_auto_install -Smeson+ninja
	cd obj-x86_64-linux-gnu && DESTDIR=/home/mick/packages/deb-create_tweaks_gtk/work_mkdeb/labwc-tweaks-gtk-0.1.0/debian/labwc-tweaks-gtk LC_ALL=C.UTF-8 ninja install
[1/11] Compiling C object labwc-tweaks.p/keyboard-layouts.c.o
[2/11] Compiling C object labwc-tweaks.p/theme.c.o
../theme.c: In function âtheme_findâ:
../theme.c:47:40: warning: â%sâ directive output may be truncated writing up to 255 bytes into a region of size between 0 and 4095 [-Wformat-truncation=]
   47 |         snprintf(buf, sizeof(buf), "%s/%s", path, dirname);
      |                                        ^~
In file included from /usr/include/stdio.h:964,
                 from ../theme.c:6:
In function âsnprintfâ,
    inlined from âisdirâ at ../theme.c:47:2,
    inlined from âadd_theme_if_icon_themeâ at ../theme.c:71:35,
    inlined from âtheme_findâ at ../theme.c:211:4:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:54:10: note: â__builtin___snprintf_chkâ output between 2 and 4352 bytes into a destination of size 4096
   54 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   55 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   56 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
../theme.c: In function âtheme_findâ:
../theme.c:76:48: warning: â%sâ directive output may be truncated writing up to 255 bytes into a region of size between 0 and 4095 [-Wformat-truncation=]
   76 |                 snprintf(buf, sizeof(buf), "%s/%s", path, entry->d_name);
      |                                                ^~
In function âsnprintfâ,
    inlined from âadd_theme_if_icon_themeâ at ../theme.c:76:3,
    inlined from âtheme_findâ at ../theme.c:211:4:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:54:10: note: â__builtin___snprintf_chkâ output between 2 and 4352 bytes into a destination of size 4096
   54 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   55 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   56 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
../theme.c: In function âtheme_findâ:
../theme.c:47:40: warning: â%sâ directive output may be truncated writing up to 255 bytes into a region of size between 0 and 4095 [-Wformat-truncation=]
   47 |         snprintf(buf, sizeof(buf), "%s/%s", path, dirname);
      |                                        ^~
In function âsnprintfâ,
    inlined from âisdirâ at ../theme.c:47:2,
    inlined from âadd_theme_if_icon_themeâ at ../theme.c:95:40,
    inlined from âtheme_findâ at ../theme.c:211:4:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:54:10: note: â__builtin___snprintf_chkâ output between 2 and 4352 bytes into a destination of size 4096
   54 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   55 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   56 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
../theme.c: In function âtheme_findâ:
../theme.c:47:40: warning: â%sâ directive output may be truncated writing up to 255 bytes into a region of size between 0 and 4095 [-Wformat-truncation=]
   47 |         snprintf(buf, sizeof(buf), "%s/%s", path, dirname);
      |                                        ^~
In function âsnprintfâ,
    inlined from âisdirâ at ../theme.c:47:2,
    inlined from âprocess_dirâ at ../theme.c:142:34,
    inlined from âtheme_findâ at ../theme.c:202:4:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:54:10: note: â__builtin___snprintf_chkâ output between 2 and 4352 bytes into a destination of size 4096
   54 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   55 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   56 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
../theme.c: In function âtheme_findâ:
../theme.c:144:56: warning: â%sâ directive output may be truncated writing up to 255 bytes into a region of size between 0 and 4095 [-Wformat-truncation=]
  144 |                         snprintf(buf, sizeof(buf), "%s/%s/%s", path, entry->d_name, filename);
      |                                                        ^~
In function âsnprintfâ,
    inlined from âprocess_dirâ at ../theme.c:144:4,
    inlined from âtheme_findâ at ../theme.c:202:4:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:54:10: note: â__builtin___snprintf_chkâ output 3 or more bytes (assuming 4353) into a destination of size 4096
   54 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   55 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   56 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
[3/11] Compiling C object labwc-tweaks.p/environment.c.o
[4/11] Compiling C object labwc-tweaks.p/xml.c.o
[5/11] Compiling C object labwc-tweaks.p/stack-appearance.c.o
[6/11] Compiling C object labwc-tweaks.p/main.c.o
[7/11] Compiling C object labwc-tweaks.p/stack-lang.c.o
[8/11] Compiling C object labwc-tweaks.p/update.c.o
[9/11] Compiling C object labwc-tweaks.p/stack-mouse.c.o
[10/11] Linking target labwc-tweaks
[10/11] Installing files.
Installing labwc-tweaks to /home/mick/packages/deb-create_tweaks_gtk/work_mkdeb/labwc-tweaks-gtk-0.1.0/debian/labwc-tweaks-gtk/usr/bin
Installing /home/mick/packages/deb-create_tweaks_gtk/work_mkdeb/labwc-tweaks-gtk-0.1.0/data/labwc-tweaks.svg to /home/mick/packages/deb-create_tweaks_gtk/work_mkdeb/labwc-tweaks-gtk-0.1.0/debian/labwc-tweaks-gtk/usr/share/icons/hicolor/scalable/apps
Installing /home/mick/packages/deb-create_tweaks_gtk/work_mkdeb/labwc-tweaks-gtk-0.1.0/data/labwc-tweaks.desktop to /home/mick/packages/deb-create_tweaks_gtk/work_mkdeb/labwc-tweaks-gtk-0.1.0/debian/labwc-tweaks-gtk/usr/share/applications
```

--- a/theme.c
+++ b/theme.c
@@ -44,7 +44,10 @@ static bool
 isdir(const char *path, const char *dirname)
 {
 	char buf[4096];
-	snprintf(buf, sizeof(buf), "%s/%s", path, dirname);
+	int ret = snprintf(buf, sizeof(buf), "%s/%s", path, dirname);
+	if (ret < 0) {
+		return false;
+	}
 	struct stat st;
 	return (!stat(buf, &st) && S_ISDIR(st.st_mode));
 }
@@ -73,7 +76,10 @@ add_theme_if_icon_theme(struct themes *themes, const char *path)
 		}
 
 		char buf[4096];
-		snprintf(buf, sizeof(buf), "%s/%s", path, entry->d_name);
+		int ret = snprintf(buf, sizeof(buf), "%s/%s", path, entry->d_name);
+		if (ret < 0) {
+			return;
+		}
 		/* filter 'hicolor' as it is not a complete icon set */
 		if (strstr(buf, "hicolor") != NULL) {
 			continue;
@@ -141,7 +147,10 @@ process_dir(struct themes *themes, const char *path, const char *filename)
 	while ((entry = readdir(dp))) {
 		if (entry->d_name[0] != '.' && isdir(path, entry->d_name)) {
 			char buf[4096];
-			snprintf(buf, sizeof(buf), "%s/%s/%s", path, entry->d_name, filename);
+			int ret = snprintf(buf, sizeof(buf), "%s/%s/%s", path, entry->d_name, filename);
+			if (ret < 0) {
+				return;
+			}
 			/* filter 'hicolor' as it is not a complete icon set */
 			if (strstr(buf, "hicolor") != NULL) {
 				continue;
@@ -183,15 +192,22 @@ void
 theme_find(struct themes *themes, const char *middle, const char *end)
 {
 	char path[4096];
+	int ret;
 	for (uint32_t i = 0; i < ARRAY_SIZE(dirs); ++i) {
 		if (dirs[i].prefix) {
 			char *prefix = getenv(dirs[i].prefix);
 			if (!prefix) {
 				continue;
 			}
-			snprintf(path, sizeof(path), "%s/%s/%s", prefix, dirs[i].path, middle);
+			ret = snprintf(path, sizeof(path), "%s/%s/%s", prefix, dirs[i].path, middle);
+			if (ret < 0) {
+				return;
+			}
 		} else {
-			snprintf(path, sizeof(path), "%s/%s", dirs[i].path, middle);
+			ret = snprintf(path, sizeof(path), "%s/%s", dirs[i].path, middle);
+			if (ret < 0) {
+				return;
+			}
 		}
 
 		if (end) {

